src/OrchardCore/OrchardCore.Setup.Abstractions/SetupContext.cs

```diff
@@ -74,5 +74,9 @@ public class SetupContext
         /// Gets or sets the errors during set up if there is.
         /// </summary>
         public IDictionary<string, string> Errors { get; set; }
+        /// <summary>
+        /// Gets additional key/value info.
+        /// </summary>
+        public IDictionary<string, string> Properties { get; } = new Dictionary<string, string>();
```

Comment:
Can it be a `<string, object>` since I don't see any reason to limit the type of parameters.

Reply:
Bonus: Could you remove all the "string" properties in this class and use the Properties dictionary to pass them? This way the context will be more generic and won't need to be changed every time we thing the setup screen needs to be changed. The properties like ShellSettings, EnabledFeatures, Recipe, Errors should remain first class properties.

Reply:
done. Please review.